File: olymp/vkcup_4/i.py
n = int(input())

m = []

for i in range(n):
    H, M = map(int, input().split(':'))
    m.append(H*60 + M)
    m.append(m[-1]+24*60)

m.sort()

ans = 0

for i in range(1, 2*n):
    if m[i] - m[i-1] > ans:
        ans = m[i] - m[i-1]

ans -= 1
print("{}:{}".format(ans // 60, ans % 60))

File: olymp/vkcup_4/i.cpp
#include <iostream>
#include <cstdio>
#include <vector>
#include <algorithm>

using namespace std;

int main(){
    int n;
    cin >> n;
    vector<int> m;
    for (int i = 0; i < n; i++) {
        int a, b;
        scanf("%d:%d", &a, &b);
        m.push_back(a*60+ b);
        m.push_back(a*60+ b+24*60);
    }
    sort(m.begin(), m.end());
    int ans = 0;
    for (int i = 1; i < m.size(); ++i) {
        if (m[i] - m[i-1] > ans) ans = m[i] -m[i-1];

    }
    ans--;
    printf("%02d:%02d\n", ans / 60, ans % 60);

}
